packages/zk-certificates/contracts/ZkKYC.sol

```diff
@@ -89,6 +89,32 @@ contract ZkKYC is Ownable {
         uint[2][2] memory b,
         uint[2] memory c,
         uint[] memory input
+    ) public view returns (bool) {
+            // dev note: if we ever use proof hash, make sure to pay attention to this truncation to uint160 as it can violate uniqueness
+        require(
+            tx.origin == address(uint160(input[INDEX_USER_ADDRESS])),
+            'transaction submitter is not authorized to use this proof'
+        );
+        return _verifyProof(a, b, c, input);
+    }
+
+    function verifyProof2(
```

Comment:
I thought the docstring for the internal _verifyProof is already enough

Reply:
`verifyProof2` is still a bad name in my opinion because people seeing it in the ABI or in another smart contract calling this function have no idea what the difference to `verifyProof` is.